### PR TITLE
Escaping the URLs in network_site_url function

### PR DESCRIPTION
**Pull request by Rajinsharwar**
Escaping the URLs in the network_site_url function in the file of wp-activate.php

Trac ticket: https://core.trac.wordpress.org/ticket/57336

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-activate.php [hunk 1]
```diff
@@ -150,10 +150,10 @@ function wpmu_activate_stylesheet() {
 				printf(
 					/* translators: 1: Login URL, 2: Username, 3: User email address, 4: Lost password URL. */
 					__( 'Your account has been activated. You may now <a href="%1$s">log in</a> to the site using your chosen username of &#8220;%2$s&#8221;. Please check your email inbox at %3$s for your password and login instructions. If you do not receive an email, please check your junk or spam folder. If you still do not receive an email within an hour, you can <a href="%4$s">reset your password</a>.' ),
-					network_site_url( $blog_details->path . 'wp-login.php', 'login' ),
-					$signup->user_login,
-					$signup->user_email,
-					wp_lostpassword_url()
+					esc_url( network_site_url( $blog_details->path . 'wp-login.php', 'login' ) ),
+					esc_html( $signup->user_login ),
+					esc_html( $signup->user_email ),
```
Comment (costdev) [inline src/wp-activate.php:155]: `esc_html()` here goes further than the PR and ticket intended. However, I'm not opposed to this PR handling the additional escaping in this file. We'd just need to adjust the PR and ticket titles, as well as handle other things that need to be escaped.

There are more URLs and dynamic strings output in this file that need to be escaped. Are you happy to update these too?

Ideally, there would be two commits in this PR:
- "Escape URLs."
- "Escape HTML element content."

This makes it easier should a committer decide they want to separate these in the final commit(s).

File: src/wp-activate.php [hunk 2]
```diff
@@ -127,10 +127,10 @@ function wpmu_activate_stylesheet() {
 	<div class="wp-activate-container">
 	<?php if ( ! $key ) { ?>
 
-		<h2><?php _e( 'Activation Key Required' ); ?></h2>
-		<form name="activateform" id="activateform" method="post" action="<?php echo network_site_url( $blog_details->path . 'wp-activate.php' ); ?>">
+		<h2><?php esc_html_e( 'Activation Key Required' ); ?></h2>
```
Comment (peterwilsoncc) [inline src/wp-activate.php]: WordPress core intentionally does not escape translations as there is a review process for the translation of strings.

```suggestion
		<h2><?php _e( 'Activation Key Required' ); ?></h2>
```

File: src/wp-activate.php [hunk 3]
```diff
@@ -127,10 +127,10 @@ function wpmu_activate_stylesheet() {
...
+		<form name="activateform" id="activateform" method="post" action="<?php echo esc_url( network_site_url( $blog_details->path . 'wp-activate.php' ) ); ?>">
 			<p>
-				<label for="key"><?php _e( 'Activation Key:' ); ?></label>
+				<label for="key"><?php esc_html_e( 'Activation Key:' ); ?></label>
```
Comment (peterwilsoncc) [inline src/wp-activate.php]: As above.

```suggestion
				<label for="key"><?php _e( 'Activation Key:' ); ?></label>
```

File: src/wp-activate.php [hunk 4]
```diff
@@ -150,38 +150,38 @@ function wpmu_activate_stylesheet() {
...
 				);
 			} else {
 				printf(
 					/* translators: 1: Site URL, 2: Username, 3: User email address, 4: Lost password URL. */
-					__( 'Your site at %1$s is active. You may now log in to your site using your chosen username of &#8220;%2$s&#8221;. Please check your email inbox at %3$s for your password and login instructions. If you do not receive an email, please check your junk or spam folder. If you still do not receive an email within an hour, you can <a href="%4$s">reset your password</a>.' ),
-					sprintf( '<a href="http://%1$s%2$s">%1$s%2$s</a>', $signup->domain, $blog_details->path ),
-					$signup->user_login,
-					$signup->user_email,
-					wp_lostpassword_url()
+					esc_html__( 'Your site at %1$s is active. You may now log in to your site using your chosen username of &#8220;%2$s&#8221;. Please check your email inbox at %3$s for your password and login instructions. If you do not receive an email, please check your junk or spam folder. If you still do not receive an email within an hour, you can <a href="%4$s">reset your password</a>.' ),
```
Comment (peterwilsoncc) [inline src/wp-activate.php]: As above.

```suggestion
					__( 'Your site at %1$s is active. You may now log in to your site using your chosen username of &#8220;%2$s&#8221;. Please check your email inbox at %3$s for your password and login instructions. If you do not receive an email, please check your junk or spam folder. If you still do not receive an email within an hour, you can <a href="%4$s">reset your password</a>.' ),
```

File: src/wp-activate.php [hunk 5]
```diff
@@ -150,38 +150,38 @@ function wpmu_activate_stylesheet() {
...
+					esc_url( sprintf( '<a href="http://%1$s%2$s">%1$s%2$s</a>', $signup->domain, $blog_details->path ) ),
```
Comment (peterwilsoncc) [inline src/wp-activate.php]:
```suggestion
					sprintf( '<a href="http://%1$s">%1$s</a>', esc_url ( $signup->domain . $blog_details->path ) ),
```

File: src/wp-activate.php [hunk 6]
```diff
@@ -150,38 +150,38 @@ function wpmu_activate_stylesheet() {
...
+					esc_html( $signup->user_login ),
+					esc_html( $signup->user_email ),
+					esc_url( wp_lostpassword_url() )
 				);
 			}
 			echo '</p>';
 		} elseif ( null === $result || is_wp_error( $result ) ) {
 			?>
-			<h2><?php _e( 'An error occurred during the activation' ); ?></h2>
+			<h2><?php esc_html_e( 'An error occurred during the activation' ); ?></h2>
```
Comment (peterwilsoncc) [inline src/wp-activate.php]:
```suggestion
			<h2><?php _e( 'An error occurred during the activation' ); ?></h2>
```

File: src/wp-activate.php [hunk 7]
```diff
@@ -150,38 +150,38 @@ function wpmu_activate_stylesheet() {
...
+				<p><?php echo esc_html( $result->get_error_message() ); ?></p>
 			<?php endif; ?>
 			<?php
 		} else {
-			$url  = isset( $result['blog_id'] ) ? get_home_url( (int) $result['blog_id'] ) : '';
+			$url  = isset( $result['blog_id'] ) ? esc_url( get_home_url( (int) $result['blog_id'] ) ) : '';
 			$user = get_userdata( (int) $result['user_id'] );
 			?>
-			<h2><?php _e( 'Your account is now active!' ); ?></h2>
+			<h2><?php esc_html_e( 'Your account is now active!' ); ?></h2>
```
Comment (peterwilsoncc) [inline src/wp-activate.php]:
```suggestion
			<h2><?php _e( 'Your account is now active!' ); ?></h2>
```

File: src/wp-activate.php [hunk 8]
```diff
@@ -150,38 +150,38 @@ function wpmu_activate_stylesheet() {
...
 
 			<div id="signup-welcome">
-			<p><span class="h3"><?php _e( 'Username:' ); ?></span> <?php echo $user->user_login; ?></p>
-			<p><span class="h3"><?php _e( 'Password:' ); ?></span> <?php echo $result['password']; ?></p>
+			<p><span class="h3"><?php esc_html_e( 'Username:' ); ?></span> <?php echo esc_html( $user->user_login ); ?></p>
+			<p><span class="h3"><?php esc_html_e( 'Password:' ); ?></span> <?php echo esc_html( $result['password'] ); ?></p>
```
Comment (peterwilsoncc) [inline src/wp-activate.php]:
```suggestion
			<p><span class="h3"><?php _e( 'Username:' ); ?></span> <?php echo esc_html( $user->user_login ); ?></p>
			<p><span class="h3"><?php _e( 'Password:' ); ?></span> <?php echo esc_html( $result['password'] ); ?></p>
```

File: src/wp-activate.php [hunk 9]
```diff
@@ -193,17 +193,17 @@ function wpmu_activate_stylesheet() {
 				<p class="view">
 				<?php
 					/* translators: 1: Site URL, 2: Login URL. */
-					printf( __( 'Your account is now activated. <a href="%1$s">View your site</a> or <a href="%2$s">Log in</a>' ), $url, esc_url( $login_url ) );
+					printf( esc_html__( 'Your account is now activated. <a href="%1$s">View your site</a> or <a href="%2$s">Log in</a>' ), $url, esc_url( $login_url ) );
```
Comment (peterwilsoncc) [inline src/wp-activate.php]:
```suggestion
					printf( __( 'Your account is now activated. <a href="%1$s">View your site</a> or <a href="%2$s">Log in</a>' ), esc_url( $url ), esc_url( $login_url ) );
```

File: src/wp-activate.php [hunk 10]
```diff
@@ -193,17 +193,17 @@ function wpmu_activate_stylesheet() {
...
 				</p>
 			<?php else : ?>
 				<p class="view">
 				<?php
 					printf(
 						/* translators: 1: Login URL, 2: Network home URL. */
-						__( 'Your account is now activated. <a href="%1$s">Log in</a> or go back to the <a href="%2$s">homepage</a>.' ),
-						network_site_url( $blog_details->path . 'wp-login.php', 'login' ),
-						network_home_url( $blog_details->path )
+						esc_html__( 'Your account is now activated. <a href="%1$s">Log in</a> or go back to the <a href="%2$s">homepage</a>.' ),
```
Comment (peterwilsoncc) [inline src/wp-activate.php]:
```suggestion
						__( 'Your account is now activated. <a href="%1$s">Log in</a> or go back to the <a href="%2$s">homepage</a>.' ),
```

**Review by peterwilsoncc — COMMENTED** [review]
I've added a few notes inline.

WordPress core intentionally does not escape translations as there is a review process for the translation of strings, so I've removed all escaping for those. In some instances these changes would cause problems with linking.

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! This was merged in [r57625](https://core.trac.wordpress.org/changeset/57625).

**Pull request closed without merging**